### PR TITLE
[RNMobile] Use original repository for some `react-native-editor` dependencies

### PR DESCRIPTION
**Pull request by fluiddot**
**Gutenberg Mobile PR:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/4641

## What?

Use the original repository for fetching some of the package dependencies within `react-native-editor` instead of the forked repository.

## Why?

This is part of an effort to stop using forked repositories for some of the `react-native-editor` dependencies. The idea is to remove the ones that only introduce logic for publishing the Android library in their forked repositories.

## How?

The following packages are now pointing to their original repository but with the same version used in their forked repository.

- `react-native-get-random-values`: `1.4.0`
- ~~`react-native-reanimated`: `2.4.1`~~ We can't use this version due to [a known issue](https://github.com/wordpress-mobile/react-native-reanimated/pull/8#issuecomment-1050689949), for now we'll stick to `2.2.0` forked version. 
- `react-native-safe-area-context`: `3.2.0`
- `react-native-screens`: `2.9.0`
- `react-native-svg`: `9.13.6`
- `react-native-webview`: `11.6.2`
- `@react-native-masked-view/masked-view`: `0.2.6`
- `@react-native-clipboard/clipboard`: `1.9.0`
- ~~`react-native-gesture-handler`: `2.2.0`~~ We'll use the forked version because we need a fix for unit tests.

Additionally, some modifications to the Android build configuration have been applied to use the dependency artifacts instead including the source code.

## Testing Instructions

Each dependency has its testing instructions.

### `react-native-get-random-values`

One of the uses of this package is related to obtaining a UUID for each block rendered, so it would be relatively easy to identify potential errors just by opening the editor and adding some blocks.

1. Open a post/editor in the app.
2. Observe that the editor is initialized properly without any errors.
3. Add some blocks and observe that work as expected.

### ~~`react-native-reanimated`~~

**NOTE: This package has finally not been updated so we don't need to test it.**

1. Apply the following patch diff that incorporates an example that uses the new APIs of the libraries:
<details><summary>Click here to display the patch diff</summary>

```diff
diff --git a/packages/edit-post/src/components/visual-editor/index.native.js b/packages/edit-post/src/components/visual-editor/index.native.js
index 886e56e493938b4b63c08199e39fc90f7accf4e5..25850c6b5cab4e0f2808064fdbd79c7c2481c0d7 100644
--- a/packages/edit-post/src/components/visual-editor/index.native.js
+++ b/packages/edit-post/src/components/visual-editor/index.native.js
@@ -6,12 +6,77 @@ import { BlockList } from '@wordpress/block-editor';
 /**
  * External dependencies
  */
-import { Keyboard } from 'react-native';
+import { Keyboard, StyleSheet, View } from 'react-native';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, {
+	useSharedValue,
+	useAnimatedStyle,
+	withSpring,
+} from 'react-native-reanimated';
 /**
  * Internal dependencies
  */
 import Header from './header';
 
+const TestComponent = () => {
+	const ballStyles = StyleSheet.create( {
+		container: {
+			position: 'absolute',
+			bottom: 64,
+			right: 32,
+			zIndex: 10,
+		},
+		ball: {
+			width: 100,
+			height: 100,
+			borderRadius: 100,
+			backgroundColor: 'blue',
+		},
+	} );
+
+	const isPressed = useSharedValue( false );
+	const offset = useSharedValue( { x: 0, y: 0 } );
+	const animatedStyles = useAnimatedStyle( () => {
+		return {
+			transform: [
+				{ translateX: offset.value.x },
+				{ translateY: offset.value.y },
+				{ scale: withSpring( isPressed.value ? 1.2 : 1 ) },
+			],
+			backgroundColor: isPressed.value ? 'yellow' : 'blue',
+		};
+	} );
+
+	const start = useSharedValue( { x: 0, y: 0 } );
+	const gesture = Gesture.Pan()
+		.onBegin( () => {
+			isPressed.value = true;
+		} )
+		.onUpdate( ( e ) => {
+			offset.value = {
+				x: e.translationX + start.value.x,
+				y: e.translationY + start.value.y,
+			};
+		} )
+		.onEnd( () => {
+			start.value = {
+				x: offset.value.x,
+				y: offset.value.y,
+			};
+		} )
+		.onFinalize( () => {
+			isPressed.value = false;
+		} );
+
+	return (
+		<View style={ ballStyles.container }>
+			<GestureDetector gesture={ gesture }>
+				<Animated.View style={ [ ballStyles.ball, animatedStyles ] } />
+			</GestureDetector>
+		</View>
+	);
+};
+
 export default class VisualEditor extends Component {
 	constructor( props ) {
 		super( props );
@@ -58,11 +123,14 @@ export default class VisualEditor extends Component {
 		const { isAutoScrollEnabled } = this.state;
 
 		return (
-			<BlockList
-				header={ this.renderHeader }
-				safeAreaBottomInset={ safeAreaBottomInset }
-				autoScroll={ isAutoScrollEnabled }
-			/>
+			<>
+				<TestComponent />
+				<BlockList
+					header={ this.renderHeader }
+					safeAreaBottomInset={ safeAreaBottomInset }
+					autoScroll={ isAutoScrollEnabled }
+				/>
+			</>
 		);
 	}
 }
```

</details>

2. Open any post/page.
3. Observe that a blue ball is displayed in the right lower corner.
4. Observe that the ball can be dragged and dropped around the screen.

### `react-native-safe-area-context`

Looks like this package is not being used or at least is not referenced anywhere in the code. Therefore, we can't provide testing instructions for it.

### `react-native-screens`

This package is used by React navigation, so we could check if navigation within the bottom sheet component works as expected.

1. Open a post/page.
2. Add a Buttons block.
3. Tap on the Button block.
4. Tap on the gear icon button to open the block settings.
5. Tap on the "Background" field.
6. Observe that it navigates to the Background screen.
7. Tap on the back arrow button.
8. Observe that it navigates back to the block settings screen.

### `react-native-svg`

This package is mainly used for rendering the icons within the editor, so we can check for example if the block icons are properly rendered.

1. Open a post/page.
9. Tap on the ➕ icon button to display the block picker.
10. Observe that the icons for each block are rendered properly.
11. In case there are new blocks, observe that a sparkles icon is displayed on the top-left corner.

### `react-native-webview`

This package is only used by the Embed block.

1. Open a post/page.
2. Add an Embed block.
3. Set a YouTube URL (i.e. `https://www.youtube.com/watch?v=ssfHW5lwFZg`).
4. Observe that the inline preview is displayed properly.

### `@react-native-masked-view/masked-view`

Looks like this package is not being used, therefore we can't provide testing instructions.

### `@react-native-clipboard/clipboard`

This package is mostly used when setting a link.

**Auto-populate links from clipboard:**
1. Open the browser and navigate to the following URL: `https://automattic.com/how-we-work/`
2. Copy the webpage link.
3. Open a post/page.
4. Write some text on a paragraph block.
12. Select a word and tap on the link format button located in the toolbar.
13. Observe that the "Link to" field is automatically populated by the previously copied link.

**Copy download link to clipboard:**
1. Open a post/page.
2. Add a File block.
3. Add a file, for example, an image.
4. Tap on the gear icon button to open the block settings.
5. Tap on the "Copy file URL" button.
6. Go to the browser and paste the link from the clipboard.
14. Observe that the pasted link is the one related to the File block.

### ~~`react-native-gesture-handler`~~

**NOTE: This package has finally not been updated so we don't need to test it.**

This package can be tested with the same testing instructions of `react-native-reanimated`.

## Screenshots or screencast 
N/A

### CODE REVIEW

**Comment by fluiddot** [comment]
`Unit Tests / Mobile (pull_request)` PR check is failing due to the same issue as outlined in https://github.com/WordPress/gutenberg/pull/39098#issuecomment-1054527848. It should be solved by either using the forked repository version or using a newer version like `2.3.1`.

File: packages/react-native-editor/package.json [hunk 1]
```diff
@@ -57,23 +57,23 @@
...
+		"react-native-gesture-handler": "2.2.0",
+		"react-native-get-random-values": "1.4.0",
 		"react-native-hr": "git+https://github.com/Riglerr/react-native-hr.git#2d01a5cf77212d100e8b99e0310cce5234f977b3",
 		"react-native-hsv-color-picker": "https://raw.githubusercontent.com/wordpress-mobile/react-native-hsv-color-picker/v1.0.1-wp-2/react-native-hsv-color-picker-1.0.1-wp-2.tgz",
 		"react-native-keyboard-aware-scroll-view": "https://raw.githubusercontent.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/v0.8.8-wp-1/react-native-keyboard-aware-scroll-view-0.8.8-wp-1.tgz",
 		"react-native-linear-gradient": "https://raw.githubusercontent.com/wordpress-mobile/react-native-linear-gradient/v2.5.6-wp-2/react-native-linear-gradient-2.5.6-wp-2.tgz",
 		"react-native-modal": "^11.10.0",
 		"react-native-prompt-android": "https://raw.githubusercontent.com/wordpress-mobile/react-native-prompt-android/v1.0.0-wp-2/react-native-prompt-android-1.0.0-wp-2.tgz",
-		"react-native-reanimated": "https://raw.githubusercontent.com/wordpress-mobile/react-native-reanimated/2.2.4-wp-1/react-native-reanimated-2.2.4-wp-1.tgz",
+		"react-native-reanimated": "2.4.1",
```
Comment (fluiddot) [inline packages/react-native-editor/package.json]: I'm going to revert this change due to [this issue identified in WordPress-Android PR](https://github.com/wordpress-mobile/WordPress-Android/pull/16083#issuecomment-1064205673).

Comment (fluiddot) [reply packages/react-native-editor/package.json]: I applied this change in https://github.com/WordPress/gutenberg/pull/39348/commits/47a7f57cee37d03c7f5b38c7d1956833618ab2dd.

File: packages/react-native-bridge/android/react-native-bridge/build.gradle [hunk 2]
```diff
@@ -81,43 +59,31 @@ dependencies {
 
     testImplementation "junit:junit:4.13"
 
+    def rnVersion = readReactNativeVersion('../../../../package.json', 'devDependencies')
+    println "react-native version for react-native-bridge: $rnVersion"
+    def packageJson = '../../../react-native-editor/package.json'
+
+    implementation "com.facebook.react:react-native:$rnVersion"
+    implementation "org.wordpress-mobile:react-native-svg:${extractPackageVersion(packageJson, 'react-native-svg', 'dependencies')}"
```
Comment (fluiddot) [inline packages/react-native-bridge/android/react-native-bridge/build.gradle:67]: Note that the dependencies that now use the original repository are using the group ID `org.wordpress-mobile`, which is where they are published via https://github.com/wordpress-mobile/react-native-libraries-publisher repository.

**Comment by gziolo** [comment]
This is great. Thank you for working on it. It should help with upgrading npm version from v6 to v8 👏🏻

**Comment by fluiddot** [comment]
> This is great. Thank you for working on it. It should help with upgrading npm version from v6 to v8 👏🏻

Yeah, we're closer to being able to upgrade to v8 🤞 , although we still have some dependencies using forked repositories. Hopefully, we'll be also able to remove those soon, but the work is still in progress.

**Review by mchowning — APPROVED** [review]
Changes look good. Thanks so much for improving this! 🙇 

Tested via https://github.com/wordpress-mobile/WordPress-Android/pull/16083 ([apk](https://dl.circleci.com/private/output/job/dfe7f501-59d9-43bc-92fd-8c4769bd0ed5/artifacts/0/Artifacts/WordPress-pr-16083-build-119585.apk)) and https://github.com/wordpress-mobile/WordPress-iOS/pull/18117 ([ipa](https://appcenter-filemanagement-distrib5ede6f06e.azureedge.net/f128cdd4-9ec0-453d-a195-497556f9d783/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=ZaPSdEGSHVH%2FHlRlD6kmQ8Mt5nIh%2BPIaQu1Bl2DJ4WE%3D&se=2022-03-16T13%3A15%3A55Z&sp=r)), and everything works great. Thanks for giving such thorough testing instructions. They made my job easy! 🙂
